### PR TITLE
Modify Scoring Apps for OS and Host layers

### LINKED ISSUE #15: Update Scoring Method to integrate OS and Host stack layers
Currently the App and CICD layers are integrated, the score of the lower layer influences the upper layer.
We need to add the integration of more layers, more specifically, the OS and Host layers.

### PR DESCRIPTION
**Pull request by michaelehab**
Fix #15

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,24 +9,54 @@ require (
 	github.com/eclipse/paho.mqtt.golang v1.3.5
 	github.com/gorilla/mux v1.8.0
 	github.com/oklog/ulid/v2 v2.0.2
-	github.com/project-alvarium/alvarium-sdk-go v0.0.0-20240328184249-0381aa4861f0
+	github.com/project-alvarium/alvarium-sdk-go v0.0.0-20240705181105-e5ec0811a099
 	go.mongodb.org/mongo-driver v1.8.4
 )
 
+require github.com/modern-go/reflect2 v1.0.2 // indirect
```
Comment (Ali-Amin) [inline go.mod:16]: Is this package needed anywhere ? I tried building without it and the build went through, let me know if I am wrong.

**Review by Ali-Amin — DISMISSED** [review]

File: pkg/documents/types.go [hunk 2]
```diff
@@ -109,23 +109,49 @@ func NewScore(dataRef string, annotations []Annotation, policy policies.DcfPolic
...
-			// Default value that penalizes the score for not having stack confidence
-			// This happens for layers that should have stack confidence only (App, OS)
-			if layer == contracts.Application || layer == contracts.Os {
-				totalTagConfidence += 0.7
-			} else {
-				totalTagConfidence += 1.0
-			}
+			totalTagFieldConfidence += tagScore.Confidence
 		}
+		// Commented the penalty to determine how it should be done
```
Comment (tsconn23) [inline pkg/documents/types.go:116]: Should comment be removed? @michaelehab

File: pkg/documents/types.go [hunk 3]
```diff
@@ -109,23 +109,49 @@ func NewScore(dataRef string, annotations []Annotation, policy policies.DcfPolic
...
+		//              totalTagConfidence += 1.0
+		//      }
+		// }
+
+		hostFieldScore, exists := hostFieldScores[a.Host]
+		if exists {
+			totalHostFieldConfidence += hostFieldScore.Confidence
+		}
+		// Commented the penalty to determine how it should be done
+		// else {
```
Comment (tsconn23) [inline pkg/documents/types.go:132]: Should comment be removed? @michaelehab

File: go.mod (hunk 1 above)
Comment (michaelehab) [reply go.mod:16]: It keeps coming whenever I run `go mod tidy`, I haven't used it directly in my code.
image

**Review by tsconn23 — APPROVED** [review]

**Pull request merged**